Minor edits from top-to-bottom readthrough (#331)
mlswg/mls-architecture: draft-ietf-mls-architecture.md

long-term credentials. Although in principle this keying material
can be reused for multiple senders, in order to provide forward secrecy
it is better for this material to be regularly refreshed so that each

> sender can use a new key, and older keys deleted.

Reviewer: ```suggestion
sender can use a new key and delete older keys.
```

I think this technically parses as-is, but seems awkward.

each user for a mix of protocol versions, ciphersuites, and end-user devices.

When a client wishes to establish a group or add clients to a group, it first

> contacts the DS to request KeyPackages for the other clients,

Reviewer: ```suggestion
contacts the DS to request a KeyPackage for each other client,
```

Emphasizes that you need a KeyPackage for each other client.

Author: I used "each of the other clients"

> contacts the DS to request KeyPackages for the other clients,

Author: ```suggestion
contacts the DS to request KeyPackages for each of the other clients,
```

prevents version downgrade attacks where an attacker would actively rewrite
messages with a lower protocol version than the messages originally offered by the
endpoints. When multiple versions of MLS are available, the negotiation protocol

> guarantees that the version agreed upon will be the most preferred version supported in

Reviewer: "Most preferred" seems to raise the question of "preferred by whom".  (The answer is "the creator", but might not want to get into that here.)  Maybe just "best version"?

Author: I don't think it guarantees that they will get the "best" version. For instance, there might be disagreement about best.  How about
"The negotiation protocol guarantees that the creator is able to select the best version out of..."

Reviewer: That works for me.

- A policy of how to protect and share the GroupInfo objects needed for
  external joins.

- A policy for when two credentials represent the same client, distinguishing
  the following two cases:
  - When there are multiple devices for a given user.

>   - When a single has signature keys -- for instance, if the device has keys corresponding to multiple
>     overlapping time periods.

Reviewer: ```suggestion
  - When a single device has multiple signature keys -- for instance, if the device has keys corresponding to multiple
    overlapping time periods.
```

Author: oops.

deniability requires further analysis.

> ### Associating a User's Clients {#associating-a-users-clients}

Reviewer: This tag shouldn't be necessary; kramdown auto-generates tags of this form based on the text of the heading.

Author: I was worried about the ' but yeah, it seems fine.
